tests/unit_tests_new/conftest.py

```diff
@@ -0,0 +1,1587 @@
+
+        return relayed_response.json()
+
+    def storage(self) -> Mapping[str, str]:
+        request = flask.request
+        with Timer() as timer:
+            relayed_response = self.relay(request)
+        # print("*****************")
+        # print("STORAGE REQUEST:")
+        # print(request.get_json())
+        # print("STORAGE RESPONSE:")
+        # print(relayed_response.status_code, relayed_response.json())
+        # print("*****************")
+
+        self.snoop_context(request, relayed_response, timer.elapsed)
+
+        return relayed_response.json()
+
+    def storage_file(self, path) -> Mapping[str, str]:
+        request = flask.request
+        with Timer() as timer:
+            relayed_response = self.relay(request)
+        # print("*****************")
+        # print("STORAGE FILE REQUEST:")
+        # print("********PATH*********")
+        # print(path)
+        # print("********ENDPATH*********")
+        # print(request.get_json())
+        # print("STORAGE FILE RESPONSE:")
+        # print(relayed_response.json())
+        # print("*****************")
+
+        self.snoop_context(request, relayed_response, timer.elapsed, path=path)
+
+        return relayed_response.json()
+
+
+@pytest.fixture(scope="function")
+def relay_server(base_url):
+    """
+    Creates a new relay server.
+    """
+
+    @contextmanager
+    def relay_server_context(inject: Optional[List[InjectedResponse]] = None):
+        _relay_server = RelayServer(base_url=base_url, inject=inject)
+        try:
+            _relay_server.start()
+            print(f"Relay server started at {_relay_server.relay_url}")
+            with unittest.mock.patch.dict(
+                os.environ,
+                {"WANDB_BASE_URL": _relay_server.relay_url},
+            ):
+                yield _relay_server
+            print(f"Stopping relay server at {_relay_server.relay_url}")
+        finally:
+            del _relay_server
+
+    return relay_server_context
+
+
+def dict_factory():
+    def helper():
+        return dict()
+
+    return helper
+
+
+@pytest.fixture(scope="function")
+def test_settings():
+    def update_test_settings(
+        extra_settings: Union[
+            dict, wandb.sdk.wandb_settings.Settings
+        ] = dict_factory()  # noqa: B008
+    ):
+        settings = wandb.Settings(
+            console="off",
+            save_code=False,
+        )
+        if isinstance(extra_settings, dict):
+            settings.update(extra_settings, source=wandb.sdk.wandb_settings.Source.BASE)
+        elif isinstance(extra_settings, wandb.sdk.wandb_settings.Settings):
+            settings.update(extra_settings)
+        settings._set_run_start_time()
+        return settings
+
+    yield update_test_settings
+
+
+@pytest.fixture(scope="function")
+def wandb_init(user, test_settings, request):
+    # mirror wandb.sdk.wandb_init.init args, overriding name and entity defaults
+    def init(
+        job_type: Optional[str] = None,
```

Comment:
This is something that will likely get out of sync.. might make sense to either turn this into **kwargs or havea a check somewhere so it will fail when args are added.....
or were you thinking that the first person who added a test would add this.

Reply:
The idea was to enforce  static checking for the arguments, but yeah... maybe we can think of some auto way to generate this...

Reply:
sounds good, no need for a change now.  just something i noticed.  I like the idea of mypy checking of our tests eventually.... it might be much more possible now!

Reply:
let's add a test that would check this fixture against the actual wandb.init() - i think we use this pattern in multiple places